### ISSUE TITLE
Explore integrating pre-commit into repo

### ISSUE THREAD

**Issue by ntjohnson1** [sandialabs/pyttb#64]
For things like linting and formatting pre-commit hooks might be easier. Need to explore the ease of opting in or opting out of pre-commit setup and if that requires additional instructions.

**Comment by dmdunla** [comment]
For the pre-community hooks, I imagine this is before each commit as setup on the dev side. There are also GitHub actions, and we could create one that verifies all the formatting and complains when it fails. That would require that devs fix it before they finalize their PRs. And all of the mechanics of verification happen on GitHub rather than be each dev.

**Comment by ntjohnson1** [comment]
Yes this would simplify (hopefully) the dev side to make sure things are compliant at least for fixes that are automated (black/isort). I already added updated to the github actions in https://github.com/sandialabs/pyttb/pull/60 to verify black/isort now that I've made the project compliant. I additionally baked the checks only piece into our pytest (which I documented in the contributors guide). IMO having a few ways to hit the same end goal isn't too complicated in this case and gives flexibility in workflow.

I just posted a duplicate note on the black/isort ticket referring to delaying the pre-commit to here since I didn't have time to verify the opt-in/opt-out mechanism. Adding the config is fairly straightforward. [Here](https://pycqa.github.io/isort/docs/configuration/pre-commit.html) are isorts instructions
